apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/internal/ApolloPlugin.kt

```diff
@@ -151,29 +151,34 @@ open class ApolloPlugin : Plugin<Project> {
         it.group = TASK_GROUP
         it.description = "Generate Apollo models for ${compilationUnit.name.capitalize()} GraphQL queries"
 
-        val sources = compilationUnit.sources()
-
-        it.graphqlFiles.setFrom(sources.graphqlFiles)
-        it.rootFolders.set(sources.rootFolders.map { it.absolutePath })
-        it.schemaFile.set(sources.schemaFile)
-
-        it.nullableValueType.set(compilationUnit.compilerParams.nullableValueType)
-        it.useSemanticNaming.set(compilationUnit.compilerParams.useSemanticNaming)
-        it.generateModelBuilder.set(compilationUnit.compilerParams.generateModelBuilder)
-        it.useJavaBeansSemanticNaming.set(compilationUnit.compilerParams.useJavaBeansSemanticNaming)
-        it.suppressRawTypesWarning.set(compilationUnit.compilerParams.suppressRawTypesWarning)
-        it.generateKotlinModels.set(compilationUnit.compilerParams.generateKotlinModels)
-        it.generateVisitorForPolymorphicDatatypes.set(compilationUnit.compilerParams.generateVisitorForPolymorphicDatatypes)
-        it.customTypeMapping.set(compilationUnit.compilerParams.customTypeMapping)
-        it.rootPackageName.set(compilationUnit.compilerParams.rootPackageName)
+        val compilerParams = compilationUnit
```

Comment:
This is nice. The rest is more clear. It is also more clear here that we do fallbacks.

Reply:
Yes, exactly. The only special case now is with `generateKotlinModels`.